transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/CommonTransformations.scala

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.monster.encode.transformation
+
+import upack.{Msg, Obj, Str}
+
+import scala.collection.mutable
+
+/** Transformation logic that applies across multiple ENCODE objects. */
+object CommonTransformations {
+  import org.broadinstitute.monster.common.msg.MsgOps
+
+  /** Message value used by ENCODE in place of null. */
+  private val UnknownValue: Msg = Str("unknown")
+
+  /**
+    * Process an arbitrary ENCODE object to strip out all
+    * values representing null / absence of data.
+    */
+  def removeUnknowns(rawObject: Msg): Msg = {
+    val cleaned = new mutable.LinkedHashMap[Msg, Msg]()
+    rawObject.obj.foreach {
+      case (_, UnknownValue) => ()
+      case (k, v)            => cleaned += k -> v
+    }
+    new Obj(cleaned)
+  }
+
+  /**
+    * Regex matching raw ENCODE @id values, capturing the accession /
+    * other ID value and dropping the entity-type prefix.
+    */
+  private val IdPattern = "/[^/]+/([^/]+)/".r("accession")
+
+  /**
+    * Transform a raw ENCODE id into our preferred format by stripping
+    * its prefix and suffix.
+    */
+  def transformId(rawId: String): String =
+    IdPattern.findAllIn(rawId).group("accession")
+
+  /**
+    * Read the @id out of a raw ENCODE object, transforming it into our
+    * preferred format along the way.
+    */
+  def readId(rawObj: Msg): String = transformId(rawObj.read[String]("@id"))
+
+  /**
+    * Summarize any audit records found in a raw ENCODE object, returning:
+    *   1. A color label for the max audit level present
+    *   2. The set of unique audit categories present
+    */
+  def summarizeAudits(rawObject: Msg): (Option[String], Array[String]) =
```

Comment:
This is used in my file branch. Didn't mean for it to sneak through, but figured it could be useful for others working on tickets that involve parsing audits.